src/JobQueue.cc

```diff
@@ -36,7 +36,7 @@ bool JobQueue::enqueuePromiseJob(JSContext *cx,
 
   // Convert the `job` JS function to a Python function for event-loop callback
   JS::RootedValue jobv(cx, JS::ObjectValue(*job));
-  PyObject *callback = pyTypeFactory(cx, jobv)->getPyObject();
+  PyObject *callback = pyTypeFactory(cx, jobv);
```

Comment:
We need to call `Py_DECREF(callback)` after `loop.enqueue(callback)` is called. (`loop.enqueue` holds its own reference on `callback`)

Reply:
done